### PR TITLE
Add editorconfig

### PR DESCRIPTION
**Pull request by bennettgoble**
Provide a basic editorconfig file with this repository's defaults for spacing and line ending styles. This can go into any branch: it should only impact new code.

### CODE REVIEW

**Review by marchcat — APPROVED** [review]
This should work great with the Maint X update from #1373.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+[*]
+charset = utf-8
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[Makefile]
+indent_style = tab
```
Comment (marchcat) [inline .editorconfig:11]: We currently use spaces for makefiles as well. At least, my script did nothing to them except trimming trailing spaces..

Comment (nat-goodspeed) [reply .editorconfig:11]: ? At least some versions of `make` infer a semantic distinction between tab indent and space indent.

Comment (marchcat) [reply .editorconfig:11]: Ah, I've been thinking about `cmake`.

**Pull request merged**